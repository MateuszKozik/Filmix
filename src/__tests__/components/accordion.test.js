import { render, fireEvent, queryByText } from "@testing-library/react";
import faqsData from "../../fixtures/faqs.json";
import { Accordion } from "../../components";

describe("<Accordion />", () => {
	it("renders the <Accordion /> with populated data", () => {
		const { container, getByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{faqsData.map((item) => (
					<Accordion.Item key={item.id}>
						<Accordion.Header>{item.header}</Accordion.Header>
						<Accordion.Body>{item.body}</Accordion.Body>
					</Accordion.Item>
				))}
			</Accordion>
		);

		expect(getByText("Frequently Asked Questions")).toBeTruthy();
		expect(getByText("What is Filmix?")).toBeTruthy();
		expect(getByText("How much does Filmix cost?")).toBeTruthy();
		expect(getByText("Where can I watch?")).toBeTruthy();
		expect(getByText("How do I cancel?")).toBeTruthy();
		expect(getByText("What can I watch on Filmix?")).toBeTruthy();
		expect(container.firstChild).toMatchSnapshot();
	});

	it("opens and closes the <Accordion /> component", () => {
		const { container, queryByText } = render(
			<Accordion>
				<Accordion.Title>Frequently Asked Questions</Accordion.Title>
				{faqsData.map((item) => (
					<Accordion.Item key={item.id}>
						<Accordion.Header>{item.header}</Accordion.Header>
						<Accordion.Body>{item.body}</Accordion.Body>
					</Accordion.Item>
				))}
			</Accordion>
		);

		const whatisFlimixBodyText =
			"Filmix is a streaming service that offers a wide variety of films and series!";

		expect(queryByText(whatisFlimixBodyText)).toBeFalsy();
		fireEvent.click(queryByText("What is Filmix?"));
		expect(queryByText(whatisFlimixBodyText)).toBeTruthy();
		fireEvent.click(queryByText("What is Filmix?"));
		expect(queryByText(whatisFlimixBodyText)).toBeFalsy();
		expect(container.firstChild).toMatchSnapshot();
	});
});
