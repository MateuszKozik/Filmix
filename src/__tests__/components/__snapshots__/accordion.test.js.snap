// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-sc-1ocdupn-0 hzlfrq"
>
  <div
    class="accordion__Inner-sc-1ocdupn-1 iOtCVp"
  >
    <h1
      class="accordion__Title-sc-1ocdupn-5 hSscJT"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        What is Filmix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        How much does Filmix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        What can I watch on Filmix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-1ocdupn-0 hzlfrq"
>
  <div
    class="accordion__Inner-sc-1ocdupn-1 iOtCVp"
  >
    <h1
      class="accordion__Title-sc-1ocdupn-5 hSscJT"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        What is Filmix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        How much does Filmix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1ocdupn-2 dsUgAB"
    >
      <div
        class="accordion__Header-sc-1ocdupn-3 gxpwDH"
      >
        What can I watch on Filmix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
